rbac/management/principal/cleaner.py

```diff
@@ -211,8 +225,17 @@ def process_principal_events_from_umb(bootstrap_service: Optional[TenantBootstra
         if not str(e).startswith(("Already connected", "Already subscribed")):
             raise e
 
-    while UMB_CLIENT.canRead(2):  # Check if queue is empty, two sec timeout
+    while UMB_CLIENT.canRead(15):  # Check if queue is empty, 15 sec timeout
         frame = UMB_CLIENT.receiveFrame()
-        process_umb_event(frame, UMB_CLIENT, bootstrap_service)
+        if not process_umb_event(frame, UMB_CLIENT, bootstrap_service):
+            break
     UMB_CLIENT.disconnect()
     logger.info("process_tenant_principal_events: Principal event processing finished.")
+
+
+def _lock_listener() -> bool:
+    """Attempt to acquire a lock for the listener and if acquired return True, else False."""
+    with connection.cursor() as cursor:
+        cursor.execute("SELECT pg_try_advisory_xact_lock(%s);", [LOCK_ID])
+        result = cursor.fetchone()
+    return result[0]  # Returns True if lock acquired, False otherwise
```

Comment:
can result be `None` ?

Reply:
Do we want to log situation when lock is not acquired ?

Reply:
Dig it! Thanks for the questions.